gossip/privdata/pvtdataprovider.go

```diff
@@ -202,27 +202,44 @@ func (pdp *PvtdataProvider) RetrievePvtdata(pvtdataToRetrieve []*ledger.TxPvtdat
 
 	pvtdata := make(rwsetByKeys)
 
+	//If there is no private data to retrieve for the block, skip all population attempts and return
+	if len(pvtdataRetrievalInfo.eligibleMissingKeys) == 0 {
+		pdp.logger.Debugf("No eligible collection private write sets to fetch for block [%d]", pdp.blockNum)
+		retrievedPvtdata.pvtdataRetrievalInfo = pvtdataRetrievalInfo
+		retrievedPvtdata.blockPvtdata = pdp.prepareBlockPvtdata(pvtdata, pvtdataRetrievalInfo)
+		return retrievedPvtdata, nil
+	}
+
+	numEligibleMissingKeysToRetrieve := len(pvtdataRetrievalInfo.eligibleMissingKeys)
+
 	// POPULATE FROM CACHE
+	numRemainingToFetch := len(pvtdataRetrievalInfo.eligibleMissingKeys)
 	pdp.populateFromCache(pvtdata, pvtdataRetrievalInfo, pvtdataToRetrieve)
+	numFetchedFromLocalCache := numRemainingToFetch - len(pvtdataRetrievalInfo.eligibleMissingKeys)
+
 	if len(pvtdataRetrievalInfo.eligibleMissingKeys) == 0 {
-		pdp.logger.Debug("No missing collection private write sets to fetch from transient store")
+		pdp.logger.Infof("Successfully fetched all %d eligible collection private write sets for block [%d] (%d from local cache, 0 from transient store, 0 from other peers)",
+			numEligibleMissingKeysToRetrieve, pdp.blockNum, numFetchedFromLocalCache)
 		retrievedPvtdata.pvtdataRetrievalInfo = pvtdataRetrievalInfo
 		retrievedPvtdata.blockPvtdata = pdp.prepareBlockPvtdata(pvtdata, pvtdataRetrievalInfo)
 		return retrievedPvtdata, nil
 	}
 
 	// POPULATE FROM TRANSIENT STORE
-	pdp.logger.Debugf("Could not find all collection private write sets in cache for block [%d]", pdp.blockNum)
-	pdp.logger.Debugf("Fetching %d collection private write sets from transient store", len(pvtdataRetrievalInfo.eligibleMissingKeys))
+	numRemainingToFetch = len(pvtdataRetrievalInfo.eligibleMissingKeys)
 	pdp.populateFromTransientStore(pvtdata, pvtdataRetrievalInfo)
+	numFetchedFromTransientStore := numRemainingToFetch - len(pvtdataRetrievalInfo.eligibleMissingKeys)
```

Comment:
Not related to this PR. Having a populate method to remove entries from `eligibleMissingKeys` is a side-effect. To check the correctness of `numRemainingToFetch` and `numFetchedFromTransientStore`, one needs to be aware that the `populateXXX()` is not only populating the pvtdata but it is also removing entries from the `pvtdataRetrievalInfo` which is a bit unnatural from the method name. At least, the `eligibleMissingKeys` could have been named as `remainingEligibleMissingKeys`.

Reply:
I agree, it feels like the main source of confusion from some of the logic around the `eligibleMissingKeys` could be avoided if it's renamed to something indicating it changes (as Senthil suggested, `remainingEligibleMissingKeys` feels appropriate here)

Reply:
done